src/vm/dllimport.cpp

```diff
@@ -5721,12 +5721,23 @@ class LoadLibErrorTracker
         m_priorityOfLastError = 0;
     }
 
-    VOID TrackErrorCode(DWORD dwLastError)
+    VOID TrackErrorCode()
     {
         LIMITED_METHOD_CONTRACT;
 
         DWORD priority;
 
+#ifdef FEATURE_PAL
+        LPWSTR lpLastError = nullptr;
```

Comment:
You aren't allocating a buffer to put the result in. You need some type of smart string so that when LoadLibErrorTracker goes out of scope, it is deallocated.

I think InlineSString<somesize> is probably what you need. Then I think you do something like OpenUTF8Buffer to get the buffer to pass in. But I do not know the VM idioms. @jkotas what is appropriate?

Reply:
You need to convert the message from UTF8 used by the PAL to UTF16. SString is a good way to do that.

Reply:
Is there documentation anywhere on SString? I'm trying to figure out the best way to use it in this instance

Reply:
The header file has comments about the different APIs: https://github.com/dotnet/coreclr/blob/master/src/inc/sstring.h

I think you want to use it like this: `SString errorMessage(SString::Utf8, ...utf8String...);`